lib/DocumentManager.php

```diff
@@ -175,4 +265,34 @@ private function getOptionsResolver($eventName)
 
         return $resolver;
     }
+
+    /**
+     * If the exception is an instanceof `DocumentManagerException` set the
+     * document manager name that threw.
+     *
+     * @param \Exception $e
+     * @param string $context
+     */
+    private function processException(\Exception $exception, $context)
```

Comment:
I feel so strong about it. I also see it the other way round, this way the `DocumentManager` must also handle all other exceptions in some way (even if it only is about rethrowing them). The other way it only has to know about the `DocumentManagerException`.